File: app/src/main/java/com/galaxy/kittymessage/SplashActivity.java
package com.galaxy.kittymessage;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

public class SplashActivity extends AppCompatActivity {

    private ProgressBar progressBar;
    private int progressStatus = 0;
    private Handler handler = new Handler();
    private static final int PERMISSION_REQUEST_CODE = 123;

    private String[] permissions = {
            Manifest.permission.READ_SMS,
            Manifest.permission.SEND_SMS,
            Manifest.permission.RECEIVE_SMS,
            Manifest.permission.READ_PHONE_STATE,
            Manifest.permission.READ_CONTACTS,
            // Add any other permissions you need
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        progressBar = findViewById(R.id.progressBar);
        progressBar.setMax(100);

        // Check if permissions are granted
        if (checkPermissions()) {
            startProgress();
        } else {
            requestPermissions();
        }
    }

    private boolean checkPermissions() {
        for (String permission : permissions) {
            if (ContextCompat.checkSelfPermission(this, permission) != PackageManager.PERMISSION_GRANTED) {
                return false;
            }
        }
        return true;
    }

    private void requestPermissions() {
        ActivityCompat.requestPermissions(this, permissions, PERMISSION_REQUEST_CODE);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);

        if (requestCode == PERMISSION_REQUEST_CODE) {
            boolean allPermissionsGranted = true;
            for (int grantResult : grantResults) {
                if (grantResult != PackageManager.PERMISSION_GRANTED) {
                    allPermissionsGranted = false;
                    break;
                }
            }

            if (allPermissionsGranted) {
                startProgress();
            } else {
                Toast.makeText(this, "Permissions not granted. Exiting app.", Toast.LENGTH_SHORT).show();
                finish();
            }
        }
    }

    private void startProgress() {
        new Thread(() -> {
            while (progressStatus < 100) {
                progressStatus += 1;

                // Update the progress on the UI thread
                handler.post(new Runnable() {
                    public void run() {
                        progressBar.setProgress(progressStatus);
                    }
                });

                try {
                    // Delay for 20 milliseconds (adjust as needed)
                    Thread.sleep(20);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
            startMainActivity();
        }).start();

    }

    private void startMainActivity() {
        Intent intent;
        if (AppHelper.GetCurrentLanguage(SplashActivity.this).equals(AppConstant.NOT_SET_KEY)) {
            intent = new Intent(SplashActivity.this, LanguageActivity.class);
        } else {
            intent = new Intent(SplashActivity.this, MessagingActivity.class);
        }
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/galaxy/kittymessage/SplashActivity.kt
package com.galaxy.kittymessage

import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.os.CountDownTimer
import android.widget.ProgressBar
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import android.Manifest
import android.util.Log


class SplashActivity : AppCompatActivity() {

    private lateinit var progressBar: ProgressBar
    private lateinit var countDownTimer: CountDownTimer
    private var count = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        Log.e("App developer", AppConstant.APP_DEVELOPER);

        progressBar = findViewById(R.id.progressBar)
        // Set the countdown timer to 10 seconds
        countDownTimer = object : CountDownTimer(2200, 20) {
            override fun onTick(millisUntilFinished: Long) {
                // Update the text view with the remaining time
                progressBar.progress = ++ count
            }

            override fun onFinish() {
                // Stop the countdown timer and reset the text view
                countDownTimer.cancel()

                // Request permission
                if (ContextCompat.checkSelfPermission(this@SplashActivity, Manifest.permission.READ_SMS)
                    != PackageManager.PERMISSION_GRANTED
                    || ContextCompat.checkSelfPermission(this@SplashActivity, Manifest.permission.SEND_SMS)
                    != PackageManager.PERMISSION_GRANTED
                    || ContextCompat.checkSelfPermission(this@SplashActivity, Manifest.permission.READ_PHONE_STATE)
                    != PackageManager.PERMISSION_GRANTED
                    || ContextCompat.checkSelfPermission(this@SplashActivity, Manifest.permission.READ_CONTACTS)
                    != PackageManager.PERMISSION_GRANTED) {
                    ActivityCompat.requestPermissions(this@SplashActivity,
                        arrayOf(Manifest.permission.READ_SMS, Manifest.permission.SEND_SMS, Manifest.permission.READ_PHONE_STATE, Manifest.permission.READ_CONTACTS),
                        1)
                } else {
                    // Permission already granted, start the main activity
                    if (AppHelper.GetCurrentLanguage(this@SplashActivity).equals("not_set")) {
                        startActivity(Intent(this@SplashActivity, LanguageActivity::class.java))
                    } else {
                        startActivity(Intent(this@SplashActivity, MessagingActivity::class.java))
                    }
                    finish()
                }
            }
        }.start()
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (grantResults.isNotEmpty()
            && grantResults[0] == PackageManager.PERMISSION_GRANTED
            && grantResults[1] == PackageManager.PERMISSION_GRANTED
            && grantResults[2] == PackageManager.PERMISSION_GRANTED
            && grantResults[3] == PackageManager.PERMISSION_GRANTED) {
            // Permission granted, start the main activity
            startActivity(Intent(this, LanguageActivity::class.java))
            finish()
        } else {
            // Permission denied, show a message to the user
            Toast.makeText(this, "Permission denied", Toast.LENGTH_SHORT).show()
        }
    }
}